app/email/email_batching.go

```diff
@@ -148,7 +159,34 @@ func (job *EmailBatchingJob) handleNewNotifications() {
 
 func (job *EmailBatchingJob) checkPendingNotifications(now time.Time, handler func(string, []*batchedNotification)) {
 	for userID, notifications := range job.pendingNotifications {
+		// Defensive code.
+		if len(notifications) == 0 {
+			continue
+		}
+
+		// get how long we need to wait to send notifications to the user
+		var interval int64
+		preference, err := job.service.store.Preference().Get(userID, model.PreferenceCategoryNotifications, model.PreferenceNameEmailInterval)
+		if err != nil {
+			// use the default batching interval if an error occurs while fetching user preferences
+			interval, _ = strconv.ParseInt(model.PreferenceEmailIntervalBatchingSeconds, 10, 64)
```

Comment:
Shall we log the error here?

Reply:
We are parsing a constant value, so it's pretty much guaranteed to be correct. :) The real issue is that it's a string, whereas it should have been an int. But I think that requires further code changes, and possibly a breaking change.

Reply:
I meant the error from this line:
```Go
preference, err := job.service.store.Preference().Get(userID, model.PreferenceCategoryNotifications, model.PreferenceNameEmailInterval)
```

Reply:
Oh I see, I think an error from that is expected if the user hasn't set a custom notification preference. But the issue is we aren't differentiating between DB failure and no rows found. We need to change the store layer accordingly. I am thinking that might get a bit out of scope. WDYT? Could we tackle that separately?